### PR TITLE
Pull request for newly created object type SPRV

### PR DESCRIPTION
**Pull request by delucapietro**
It seems that no example was created by report SAFF_GENERATE_REPO.

### CODE REVIEW

File: sprv/type/zif_aff_sprv_v1.intf.abap [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    "! Create a service provider
+    BEGIN OF ty_provdetails,
+      "! <p class="shorttext">Name</p>
+      "! Name of the Service Provider
+      "! $required
+      name TYPE c LENGTH 30,
+      "! <p class="shorttext">Original Language</p>
+      "! Original Language
+      "! $showAlways
+      org_lang TYPE c LENGTH 2,
```
Comment (schneidermic0) [inline sprv/type/zif_aff_sprv_v1.intf.abap:16]: Why do you need the original language again? Is it different than the original language in the header?

File: sprv/type/zif_aff_sprv_v1.intf.abap [hunk 2]
```diff
@@ -0,0 +1,37 @@
+interface zif_aff_sprv_v1
+public .
+
+
+  types:
+    "! <p class="shorttext">Create a service provider</p>
+    "! Create a service provider
```
Comment (schneidermic0) [inline sprv/type/zif_aff_sprv_v1.intf.abap:7]: Was "Create a ..." intended in front of "service provider"?

General remark: The "shorttext" shall be written in title style, the description below in sentence style.

File: sprv/type/zif_aff_sprv_v1.intf.abap [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      "! $required
+      format_version TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header TYPE zif_aff_types_v1=>ty_header_60_cloud,
+      "! <p class="shorttext">Service Provider</p>
+      "! Service Provider
+      "! $required
+      provider_details TYPE ty_provdetails,
```
Comment (schneidermic0) [inline sprv/type/zif_aff_sprv_v1.intf.abap:35]: The title for the field is "Service Provider", this typically indicates that you might want to name the field in the same way. Does it make sense to rename `provider_details` to `service_provider`.

**Review by schneidermic0 — COMMENTED** [review]
Thanks @delucapietro for your contribution. I have added some (first) questions

File: sprv/type/zif_aff_sprv_v1.intf.abap (hunk 1 above)
Comment (delucapietro) [reply sprv/type/zif_aff_sprv_v1.intf.abap:16]: You are totally right this is redundant. I will remove the original language outside the header.

**Comment by schneidermic0** [comment]
> It seems that no example was created by report SAFF_GENERATE_REPO.

I'll check with the colleagues what might be wrong here

**Comment by huber-nicolas** [comment]
@delucapietro 
The report does not create the example object for you. You have to create it on your own manually beforehand.
The report then only serializes the example object (into its json AFF format) and adds it to the github folder.

**Comment by delucapietro** [comment]
> @delucapietro The report does not create the example object for you. You have to create it on your own manually beforehand. The report then only serializes the example object (into its json AFF format) and adds it to the github folder.

@huber-nicolas 
Is there any Guide Line on how to do this? Since it is a completely new object there is no entry in the creation option. I just can add an entry to the header table without the generated subobjects.

File: file-formats/sprv/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# SPRV File Format
+
+File | Cardinality | Definition | Schema | Example
+:--- | :---  | :--- | :--- | :---
+`<name>.sprv.json` | 1 | [`zif_aff_sprv_v1.intf.abap`](./type/zif_aff_sprv_v1.intf.abap) | [`sprv-v1.json`](./sprv-v1.json) | [`z_aff_example_sprv.sprv.json`](./examples/z_aff_example_sprv.sprv.json)
```
Comment (huber-nicolas) [inline file-formats/sprv/README.md:5]:
```suggestion
`<name>.sprv.json` | 1 | [`zif_aff_sprv_v1.intf.abap`](./type/zif_aff_sprv_v1.intf.abap) | [`sprv-v1.json`](./sprv-v1.json) | 
```
Can you please remove the link to the example object.

File: file-formats/sprv/type/zif_aff_sprv_v1.intf.abap [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      prefix TYPE c LENGTH 10,
+    END OF ty_servprovider.
+  TYPES:
+    "! <p class="shorttext">SOAP Web Service Provider Model</p>
+    "! SOAP web service provider model
+    BEGIN OF ty_main,
+      "! $required
+      format_version   TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header Data
```
Comment (wurzka) [inline file-formats/sprv/type/zif_aff_sprv_v1.intf.abap:25]:
```suggestion
      "! Header data
```

Comment (wurzka) [reply file-formats/sprv/type/zif_aff_sprv_v1.intf.abap:25]: Or just `Header` as many other object types use

**Review by wurzka — CHANGES_REQUESTED** [review]
Just one title case issue

**Comment by huber-nicolas** [comment]
I made an issue as a reminder to upload the example object later. See issue #370 
So we can merge this pull request and upload the example later

**Review by wurzka — APPROVED** [review]

**Comment by albertmink** [comment]
Let me know when you want to merge. The `Check Markdown links` is not related to the pull request, so don't bother with that.

**Review by huber-nicolas — APPROVED** [review]

**Comment by huber-nicolas** [comment]
> Let me know when you want to merge. The `Check Markdown links` is not related to the pull request, so don't bother with that.

You can merge now

**Pull request merged**